drivers/include/nrf24l01p.h

```diff
@@ -1,627 +1,1366 @@
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 0 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p0 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 1 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p1 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 2 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p2 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 3 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p3 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 4 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p4 :    5;
+    /**
+     * @brief   Current number of padding bytes to configure
+     *          the payload width of pipe 5 (Only used if protocol is SB)
+     */
+    uint8_t cfg_plw_padd_p5 :    5;
+} nrf24l01p_cfg_t;
+
+/**
+ * @brief Struct of NRF24L01P initialization parameters
+ */
+typedef struct {
+    spi_t spi;          /**< SPI bus */
+    spi_clk_t spi_clk;  /**< SPI clock speed */
+    gpio_t pin_cs;      /**< SPI chip select gpio pin */
+    gpio_t pin_ce;      /**< NRF24L01P chip enable gpio pin */
+    gpio_t pin_irq;     /**< NRF24L01P IRQ gpio pin */
+    union {
+        struct {
+            /**
+             * @brief   Array to access the addresses of pipe 0 and pipe 1 via
+             *          pipe indices @see nrf24l01p_pipe_t
+             */
+            uint8_t rx_addr_long[2][NRF24L01P_MAX_ADDR_WIDTH];
+            /**
+             * @brief   Array to access the addresses of pipe 2, pipe 3,
+             *          pipe 4 and pipe 5 via pipe indices
+             *          @see nrf24l01p_pipe_t
+             */
+            uint8_t rx_addr_short[4];
+        } arxaddr; /**< Rx addresses as arrays */
+        struct {
+            /**
+             * @brief   pipe 0 Rx address
+             */
+            uint8_t rx_pipe_0_addr[NRF24L01P_MAX_ADDR_WIDTH];
+            /**
+             * @brief   pipe 1 Rx address
+             */
+            uint8_t rx_pipe_1_addr[NRF24L01P_MAX_ADDR_WIDTH];
+            /**
+             * @brief   pipe 2 Rx address
+             */
+            uint8_t rx_pipe_2_addr;
+            /**
+             * @brief   pipe 3 Rx address
+             */
+            uint8_t rx_pipe_3_addr;
+            /**
+             * @brief   pipe 4 Rx address
+             */
+            uint8_t rx_pipe_4_addr;
+            /**
+             * @brief   pipe 5 Rx address
+             */
+            uint8_t rx_pipe_5_addr;
+        } rxaddrpx; /**< Rx addresses as named variables */
+    } urxaddr; /**< Union of Rx addresses as named variables and as arrays */
```

Comment:
If there is no advantage in having two ways to access the addresses, I'd say the `union` makes reading the code harder. The struct named `rxaddrpx` is pretty easy to understand.

Reply:
The advantage in having the addresses also in an array structure is e.g. in
`nrf24l01p_set_rx_address()` in `nrf24l01p.h`. The user passes the pipe index `pipe`. If I did not have the array, I would have to write something like:
```C
if (pipe == NRF24L01P_P0) {...}
else if (pipe == NRF24L01P_P1 {...}
else if (pipe == NRF24L01P_P2 {...}
...
```
Since NRF24L01P_Pi maps to i, I can access the address via the index.